adapters/mqtt-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1546,6 +1546,16 @@ private Future<ProtocolAdapterCommandConsumer> createCommandConsumer(final Comma
                                 subscription.getDeviceId(),
                                 authenticatedDevice,
                                 commandContext.getTracingContext());
+                    } else {
+                        // in case of gateway having subscribed for itself as simple device do not forward messages for
+                        // devices behind that gateway
+                        if (!subscription.isGatewaySubscriptionForAllDevices() && command.isTargetedAtGateway()) {
+                            log.debug(
+                                    "no command subscription via gateway found [tenant: {}, device-id: {}, gateway-id: {}]",
+                                    command.getTenant(), command.getDeviceId(), command.getGatewayId());
+                            return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_NOT_FOUND,
+                                    "no command subscription via gateway"));
```

Comment:
I think the exception message here and the log message above could be improved.
Something like "Mapped target gateway not subscribed to commands for devices behind that gateway", or "Mapped target gateway only subscribed to commands for itself".

Reply:
done